test/unit/const_string.cpp

```diff
@@ -148,6 +197,29 @@ struct const_string_test
                 BOOST_TEST(s == "abc");
                 BOOST_TEST(! s.empty());
             }
+            {
+                static_pool<100> sp;
+                const_string s;
+                {
+                    factory a(sp.allocator());
+                    s = a(10,
+                        [](std::size_t, char* dest)
+                        {
+                            dest[0] = 'a';
+                            dest[1] = 'b';
+                            dest[2] = 'c';
+                            dest[3] = 'd';
+                            dest[4] = 'e';
+                            dest[5] = 'f';
+                            dest[6] = 'g';
+                            dest[7] = 'h';
+                            dest[8] = 'i';
+                            dest[9] = 'j';
```

Comment:
why not just
```
std::memcpy( dest, "abcdefghij", 10 );
```
?

Reply:
I was trying to follow a pattern but you're right.